src/Microsoft.DotNet.Tools.Publish/Program.cs

```diff
@@ -34,13 +35,16 @@ public static int Main(string[] args)
 
             app.OnExecute(() =>
             {
-                if (!CheckArg(framework))
+                if (framework.HasValue() && NuGetFramework.Parse(framework.Value()).IsUnsupported)
```

Comment:
Looks like you're parsing the framework twice, once here and once in `GetMatchingProjectContexts`. Sounds like we should keep the parsed value around ;)